crates/ruff/src/rules/pycodestyle/rules/tab_indentation.rs

```diff
@@ -37,17 +38,39 @@ impl Violation for TabIndentation {
 }
 
 /// W191
-pub(crate) fn tab_indentation(line: &Line, indexer: &Indexer) -> Option<Diagnostic> {
-    let indent = leading_indentation(line);
-    if let Some(tab_index) = indent.find('\t') {
-        // If the tab character is within a multi-line string, abort.
-        let tab_offset = line.start() + TextSize::try_from(tab_index).unwrap();
-        if indexer.triple_quoted_string_range(tab_offset).is_none() {
-            return Some(Diagnostic::new(
-                TabIndentation,
-                TextRange::at(line.start(), indent.text_len()),
-            ));
+pub(crate) fn tab_indentation(
+    diagnostics: &mut Vec<Diagnostic>,
+    tokens: &[LexResult],
+    locator: &Locator,
+    indexer: &Indexer,
+) {
+    for (tok, range) in tokens.iter().flatten() {
+        if matches!(tok, Tok::Newline | Tok::NonLogicalNewline) {
+            tab_indentation_at_line_start(diagnostics, locator, range.end());
         }
     }
-    None
+
+    for continuation_line in indexer.continuation_line_starts() {
+        tab_indentation_at_line_start(
+            diagnostics,
+            locator,
+            locator.full_line_end(*continuation_line),
+        );
+    }
```

Comment:
Can we add a comment why this is necessary? I understand is that the lexer emits a `Newline` or `NonLogicalNewline` token for every `\n`. Is this to capture tab indentation inside multiline strings?

Reply:
The lexer doesn't emit `Newline` / `NonLogicalNewline` for a line continuation. So, for the following code snippet where there's a tab indentation before the `or` token:

```python
x \
	or y
```

The lexer will emit:
```
Name { name: "x" }
Or
Name { name: "y" }
Newline
```

This is to capture tab indentation for multi-line expression separated using a line continuation character.

Reply:
What happens if you have a continuation within a single-quoted or triple-quoted string?

Reply:
We don't check for tab indentation inside a string and the lexer captures the entire string as a single token.

Reply:
So, I guess an example like:
```python
	"""first line (tab indented start quoted)
	<-- tab indentation
"""

would only emit a diagnostic for the first line and not the second line as that's part of the string.

Oh, wait the new logic wouldn't detect the first line as there's no newline before that.

Reply:
We'll have to special case the first line to be checked every single time.

Reply:
Ok, I've updated to always check for the first line and added a test case for it.